gateway/src/shard/processor.rs

```diff
@@ -181,6 +206,7 @@ impl ShardProcessor {
 
     async fn reconnect(&mut self) {
         loop {
+            self.inflater.reset();
```

Comment:
We should be able to avoid this call because we're swapping out `self` later down, once `reconnect` is called then the inflater will never be used again.